sdk/storage/azure-storage-common/pom.xml

```diff
@@ -125,6 +125,13 @@
       <version>2.4-M1-groovy-4.0</version> <!-- {x-version-update;org.spockframework:spock-core;external_dependency} -->
       <scope>test</scope>
     </dependency>
+    <!--The Groovy version that Spock Core brings in (4.0.6) does not support Java 21, so we need to override it with a version that does.-->
```

Comment:
Could this be just an update to spock-core instead of adding the new dependency?

Reply:
No. We needed to add this because [the latest version](https://mvnrepository.com/artifact/org.spockframework/spock-core/2.4-M1-groovy-4.0) of Spock Core uses Groovy `4.0.6` :(